### PR TITLE
Simplify the core Docker image to reduce size

### LINKED ISSUE #802: Simplify Docker image
The Docker image currently contains some packages that are not necessary to run Atomix. The core image should be reduced to only the bare minimum necessary features. After all, that's how Docker is intended to be used.

### PR DESCRIPTION
**Pull request by kuujo**
This came from the Sona project:
https://github.com/sonaproject/atomix-docker
https://hub.docker.com/r/opensona/atomix-docker/tags/

Fixes #802 

Need to test the image with https://github.com/atomix/atomix-test before merging

### CODE REVIEW

**Comment by kuujo** [comment]
The Dockerfile is also missing the license/copyright header

File: dist/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-RUN mkdir /atomix
+MAINTAINER  Jordan Halterman <jordan@opennetworking.org>
 
 ARG VERSION
-COPY target/atomix-agent-${VERSION}.tar.gz /atomix/atomix-agent-${VERSION}.tar.gz
-RUN tar -xvf /atomix/atomix-agent-${VERSION}.tar.gz -C /atomix && rm /atomix/atomix-agent-${VERSION}.tar.gz
+RUN mkdir -p /root/atomix
+COPY target/atomix-agent-${VERSION}.tar.gz /root/atomix/atomix-agent-${VERSION}.tar.gz
+RUN tar -xvf /root/atomix/atomix-agent-${VERSION}.tar.gz -C /root/atomix
+RUN rm /root/atomix/atomix-agent-${VERSION}.tar.gz
```
Comment (johnou) [inline dist/Dockerfile]: better to chain both of these run commands with && to reduce the amount of layers.

Comment (kuujo) [reply dist/Dockerfile]: Ahh yes I knew there was some reason to do that

File: dist/Dockerfile [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 
-FROM  java:8
+FROM anapsix/alpine-java:8_server-jre
 
-MAINTAINER  Jordan Halterman <jordan.halterman@gmail.com>
-
-RUN apt-get update && apt-get install -y iptables git stress
-
-RUN mkdir /atomix
+MAINTAINER  Jordan Halterman <jordan@opennetworking.org>
```
Comment (jplock) [inline dist/Dockerfile]: Specifying the maintainer is deprecated

File: dist/Dockerfile [hunk 3]
```diff
@@ -1,18 +1,18 @@
 # Dockerfile
 
-FROM  java:8
+FROM anapsix/alpine-java:8_server-jre
```
Comment (jplock) [inline dist/Dockerfile]: Why use this image as a base instead of one of the officially supported openjdk alpine base images?

Comment (johnou) [reply dist/Dockerfile]: also keep in mind that if you decide to use ssl in the tcp connection the JDK implementation generates a lot of garbage and there are no alpine artifacts for netty-tcnative.

Comment (johnou) [reply dist/Dockerfile]: @kuujo try `FROM openjdk/8-jdk-slim` compressed size is 91 M.

https://hub.docker.com/r/library/openjdk/tags/

Comment (johnou) [reply dist/Dockerfile]: @kuujo try `FROM openjdk/8-jdk-alpine` it's an official image.

**Pull request merged**